homeassistant/components/onkyo/config_flow.py

```diff
@@ -165,6 +167,50 @@ async def async_step_eiscp_discovery(
             ),
         )
 
+    async def async_step_ssdp(
+        self, discovery_info: ssdp.SsdpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle flow initialized by SSDP discovery."""
+        _LOGGER.debug("Config flow start ssdp: %s", discovery_info)
+
+        udn = discovery_info.ssdp_udn
+        if udn:
+            udn_parts = udn.split(":")
+            if len(udn_parts) == 2:
+                uuid = udn_parts[1]
+                last_uuid_section = uuid.split("-")[-1].upper()
+                await self.async_set_unique_id(last_uuid_section)
+                self._abort_if_unique_id_configured()
+
+        if discovery_info.ssdp_location is None:
+            _LOGGER.error("SSDP location is None")
+            return self.async_abort(reason="unknown")
+
+        host = urlparse(discovery_info.ssdp_location).hostname
+
+        if host is None:
+            _LOGGER.error("SSDP host is None")
+            return self.async_abort(reason="unknown")
+
+        try:
+            info = await async_interview(host)
+        except Exception:
```

Comment:
Generally we don't want broad exception catches. Please narrow this to only expected exceptions.

Reply:
We have that in other places too. I think it was added to be extra defensive, but in general we don't expect it to fail.

Maybe if `loop.create_datagram_endpoint` fails or something like that.

Reply:
Catching `OSError` for `loop.create_datagram_endpoint` makes sense.  Probably anything else should raise unless there is some other expected exception as we don't want to hide bugs

Reply:
Updated to only catch OSError